docs/sources/flow/reference/config-blocks/export.md

```diff
@@ -0,0 +1,60 @@
+---
+title: export
+---
+
+# export block
+
+`export` is an optional configuration block used to specify an emitted value of
+a [Module][Modules]. `export` blocks must be given a label which determine the
+name of the export.
+
+The `export` block may not be specified in the main configuration file given
+to Grafana Agent Flow.
+
+[Modules]: {{< relref "../../concepts/modules.md" >}}
+
+## Example
+
+```river
+export "ARGUMENT_NAME" {
+  value = ARGUMENT_VALUE
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`value` | `any` | Value to export. | yes
+
+The `value` argument determines what the value of the export will be. To expose
+an exported field of another component to the module loader, set `value` to an
+expression which references that exported value.
+
+## Exported fields
+
+The `export` block does not export any fields.
+
+## Example
+
+This example creates a module where the output of discovering Kubernetes pods
+and nodes are exposed to the module loader:
+
+```river
```

Comment:
it may start to get slightly redundant, but I feel like we should have a 2nd config showing how to reference the export in the parent config for an isolated example of using an export end to end

Reply:
> but I feel like we should have a 2nd config showing how to reference the export in the parent config for an isolated example of using an export end to end

I agree, but part of my concern here is that we'll have different module loaders which do things differently. I'm not sure if we're able to commit to a generic example right now, and we'll have to push some of the responsibility for usage to the component documentation per loader.

Reply:
For example, I'd like to see a `module.kubernetes` loader which can discover FlowModule CRDs on Kubernetes and run them, but those (probably) won't support export blocks at all since the module loader will find a dynamic number of modules.